resources/documentation/learn_ansible.rst

```diff
@@ -1178,6 +1193,186 @@ And again:
 
 Handler was activated.
 
+Other useful modules
+^^^^^^^^^^^^^^^^^^^^
+
+A full list of all Ansible modules can be found on the
+`Ansible documentation <https://docs.ansible.com/ansible/latest/modules/list_of_all_modules.html>`_ .
+
+However, lets review few other often used modules:
+
+Package
+"""""""
+
+This module simply install a list of package. It can be used instead of the yum
+or dnf modules, as it covers both.
+
+Example:
+
+.. code-block:: yaml
+
+  - name: Install wget package
+    package:
+      name: wget
+      state: present
+
+  - name: Install a list of packages
+    package:
+      name: "{{ my_packages_list }}"
+      state: present
+
+.. seealso::
+  https://docs.ansible.com/ansible/latest/modules/package_module.html
+
+Service
+"""""""
+
+Manage services, to start/stop/restart, enable/disable.
+
+Example:
+
+.. code-block:: yaml
+
+  - name: Start and enable httpd service
+    service:
+      name: httpd
+      enabled: yes
+      state: started
+
+.. seealso::
+  https://docs.ansible.com/ansible/latest/modules/service_module.html
+
+File
+""""
+
+This modules allow to create folders.
+
+Example:
+
+.. code-block:: yaml
+
+  - name: Create custom folders
+    file:
+      path: "{{ item }}"
+      state: directory
+      mode: 0755
+    loop:
+     - /etc/my_folder
+     - /var/lib/my_other_folder
+
+.. seealso::
+  https://docs.ansible.com/ansible/latest/modules/file_module.html
+
+Copy
+""""
+
+Act like template, but simply copy file, no rendering.
+Files to be copied must be put into a **files** folder, at root of the role.
+
+For example, create file my_role/files/slurm.conf
+
+.. code-block:: yaml
+
+  - name: Copy slurm static file
+    copy:
+      src: slurm.conf
+      dest: /etc/slurm/slurm.conf
+      mode: 0644
+      owner: slurm
+      group: slurm
+
+.. seealso::
+  https://docs.ansible.com/ansible/latest/modules/copy_module.html
+
+Lineinfile
+""""""""""
+
+This module allows to manipulate files, using regex. For example, set a value to
+0 in a file.
+
+Example:
+
+.. code-block:: yaml
+
+  - name: Add mymanagement string to localhost in /etc/hosts
+    lineinfile:
+      path: /etc/hosts
+      regexp: '^127\.0\.0\.1'
+      line: 127.0.0.1 localhost mymanagement
+      owner: root
+      group: root
+      mode: '0644'
```

Comment:
Is it good practice to change owner/group/mode with lineinfile?
Most of the time we only want to update the content of an existing file.

Reply:
Yup, removed.